akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -628,6 +644,24 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](dest: Array[B], start: Int, len: Int): Int = {
+      if (isCompact) bytestrings.head.copyToArray(dest, start, len)
+      else {
+        // min of the bytes available top copy, bytes there is room for in dest and the requested number of bytes
+        val copied = math.max(math.min(math.min(len, length), dest.length - start), 0)
+        if (copied > 0) {
+          val bsIterator = bytestrings.iterator
+          var pos = 0
+          while (pos < copied) {
+            val current = bsIterator.next()
+            val copied = current.copyToArray(dest, pos, len)
```

Comment:
perhaps not use the name 'copied' twice ;)

also
```suggestion
            val copied = current.copyToArray(dest, pos, len - pos)
```
?